.github/workflows/CI.yaml

```diff
@@ -0,0 +1,33 @@
+name: CI
+
+env:
+  DOCKER_BUILDKIT: 1
+  COMPOSE_DOCKER_CLI_BUILD: 1
+
+on:
+  - push
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```

Comment:
@renceInbox can we use the latest `actions/checkout`, which is `v3` here? It is used further down already.